src/libs/blueprint/conduit_blueprint_mesh_examples.cpp

```diff
@@ -2532,6 +2532,389 @@ void spiral(index_t ndoms,
     }
 }
 
+//---------------------------------------------------------------------------//
+void venn_full_matset(Node &res,
+                      index_t nx,
+                      index_t ny)
+{
+    // create the material sets
+
+    index_t elements = nx * ny;
+
+    res["matsets/matset/topology"] = "topo";
+    res["matsets/matset/volume_fractions/background"] = DataType::float64(elements);
+    res["matsets/matset/volume_fractions/circle_a"] = DataType::float64(elements);
+    res["matsets/matset/volume_fractions/circle_b"] = DataType::float64(elements);
+    res["matsets/matset/volume_fractions/circle_c"] = DataType::float64(elements);
+
+    float64_array cir_a = res["fields/circle_a/values"].value();
+    float64_array cir_b = res["fields/circle_b/values"].value();
+    float64_array cir_c = res["fields/circle_c/values"].value();
+
+    float64_array mat_bg = res["matsets/matset/volume_fractions/background"].value();
+    float64_array mat_ca = res["matsets/matset/volume_fractions/circle_a"].value();
+    float64_array mat_cb = res["matsets/matset/volume_fractions/circle_b"].value();
+    float64_array mat_cc = res["matsets/matset/volume_fractions/circle_c"].value();
+    
+    for(index_t idx = 0; idx < elements; idx++)
+    {
+        mat_ca[idx] = cir_a[idx];
+        mat_cb[idx] = cir_b[idx];
+        mat_cc[idx] = cir_c[idx];
+        mat_bg[idx] = 1. - (cir_a[idx] + cir_b[idx] + cir_c[idx]);
+    }
+}
+
+void make_sparse(Node & src, index_t len, Node & n)
```

Comment:
Instead of passing `len`, couldn't we just use `src.dtype().number_of_elements()` instead? This would simplify the function signature and make the intent of the function a bit clearer. Also, naming it something like `nsrc` or `nfull` to make it crystal clear which length its describing?

Reply:
I'll get `len` from the dtype --- that is much cleaner.  I'll also rename `n` to something explicit.

Reply:
Also, I pass `nx` and `ny` all over the place.  Is there a way I can easily pull those out of a dtype?

Reply:
Not individually, but you can eliminate them from all the `venn_*` functions and replace the calculation for the `elements` variable with `res["fields/circle_a/values"].dtype().number_of_elements()`. It may be more proper to calculate this value directly from the topology itself, but just grabbing it from a spanning field (e.g. any `circle_*` field) seems fine to me as well.

Reply:
I put `nx` and `ny` into another Node (commented as "not part of the blueprint") and pull them out every time I need them.